woocommerce/woocommerce-ios
# Display a banner for onboarding user to WCShip installation in Order Detail

Closes #6914 

### Description
The first task of the Shipping Label Onboarding project is to implement a banner in the Order Detail that will be shown under certain conditions, inviting the user to install the WCShip extension.

In particular, these conditions are:
- WCShip is not installed or not enabled.
- The store is located in US.
- The currency is USD.
- The order is not eligible for IPP.

There was also another condition discussed with the team: `The products are eligible for SL (not virtual/downloadable)`. After looking at the implementation on how we check locally if the products are not virtual or downloadable, I encountered that this is not the ideal solution since during the first launch of the app, we don't have the information related to the products (that are downloaded elsewhere, and stored locally), because we fetch them from Core Data. This means that this solution is not reliable. So, since we will check at the end of the installation process if the order is eligible for shipping labels (using the endpoint under the WCShip extension), we are going to use this way to propose to the user to create a shipping label for the specific order.

### Testing instructions
1. Launch the app, and select a website without the WCShip extension installed (you can create a Jurassic ninja website, like a did). The feature flag `shippingLabelsOnboardingM1` should be enabled (enabled by default in debug).
2. Open the Order List.
3. Select an order eligible for Shipping Labels.
4. You should be able to see the banner after the products list.
5. By tapping on the banner, nothing should happen. 


### Screenshots
| Light mode             |  Dark mode |
:-------------------------:|:-------------------------:
![Simulator Screen Shot - iPhone 13 Pro Max - 2022-05-31 at 15 27 04](https://user-images.githubusercontent.com/495617/171189459-f9939de3-6b89-4d43-8eb5-c8441b63f6d7.png) | ![Simulator Screen Shot - iPhone 13 Pro Max - 2022-05-31 at 15 27 23](https://user-images.githubusercontent.com/495617/171189499-646f1aa0-55fc-4680-bcca-0726bbf0421d.png)



---
